nengo_loihi/builder/node.py

```diff
@@ -1,7 +1,7 @@
 from nengo import Node
 
 from nengo_loihi.builder.builder import Builder
-from nengo_loihi.inputs import ChipReceiveNode, SpikeInput
+from nengo_loihi.builder.inputs import ChipReceiveNode, SpikeInput
```

Comment:
Yeah I was thinking about separating out that file as well. Currently `SpikeInput` is the only one that is used from outside the `/builder/` directory. In particular, `hardware/builder.py` and `emulator/tests/test_interface.py` import `SpikeInput`, and the rest are contained within the `/builder/`.

```$ grep "from nengo_loihi.builder.inputs" . -r
./nengo_loihi/builder/node.py:from nengo_loihi.builder.inputs import ChipReceiveNode, SpikeInput
./nengo_loihi/builder/connection.py:from nengo_loihi.builder.inputs import (
./nengo_loihi/builder/builder.py:from nengo_loihi.builder.inputs import LoihiInput
./nengo_loihi/builder/tests/test_inputs.py:from nengo_loihi.builder.inputs import ChipReceiveNode, PESModulatoryTarget
./nengo_loihi/conv.py:from nengo_loihi.builder.inputs import ChipReceiveNeurons, LoihiInput
./nengo_loihi/tests/test_simulator.py:from nengo_loihi.builder.inputs import SpikeInput
./nengo_loihi/tests/test_block.py:from nengo_loihi.builder.inputs import SpikeInput
./nengo_loihi/hardware/builder.py:from nengo_loihi.builder.inputs import SpikeInput
./nengo_loihi/emulator/tests/test_interface.py:from nengo_loihi.builder.inputs import SpikeInput
```

One exception is `conv.py`, which imports `ChipReceiveNeurons` and `LoihiInput` to use within `build_conv2d_connection`. However, there is a TODO in `builder/connection.py::build_chip_connection` next to the call to `build_conv2d_connection` that says "TODO: integrate these into the same function" and so this is supposed to be in `/builder/` anyway.

What if I were to move `LoihiInput` and `SpikeInput` to `block.py`?

Reply:
If we're leaving the builder-specific ones in `builder/`, then, `LoihiInput` and `SpikeInput` *could* go in `block.py`. I'm trying to predict whether we'll have more in the near future, or not, because that would be the main thing that would motivate having `inputs.py`. The main potential one I can see on the horizon is `DVSInput`.

I think I'd vote to keep `LoihiInput` and `SpikeInput` in `inputs.py`, and move the others to `builder/inputs.py`.

Also, we should move the conv connection builder to `builder/connection.py` I think.